### PR TITLE
Add a guide on dealing with errors

### LINKED ISSUE #751: Examples for "Errors" docs
Currently https://strawberry.rocks/docs/features/errors is an empty page.

I'm happy to include some examples but, I wanted to ask, which kind(s) of errors should this page describe?

* GraphQL server erros, e.g. invalid query syntax
* Resolver errors, e.g. a `strawberry.mutation` that performs an SQL `INSERT` which fails a DB constraint

Perhaps you intend to document some data class-based pattern like the following:

```
type User {
  id: ID!
  login: String!
}

type UserNotFoundError {
  message: String!
}

union UserResult = User | UserNotFoundError

type Query {
  user(id: ID!): UserResult!
}
```

## Refs

* https://graphql-ruby.org/mutations/mutation_errors
* https://www.apollographql.com/blog/full-stack-error-handling-with-graphql-apollo-5c12da407210/

### PR DESCRIPTION
**Pull request by jkimbo**
## Description

This PR adds a guide on how to deal with the different types of errors that you will encounter with a GraphQL server.

It also renames `features` to `guides`.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Documentation

## Issues Fixed or Closed by This PR

* #764
* Closes #751

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the CONTRIBUTING document.
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by botberry** [comment]
Hi, thanks for contributing to Strawberry 🍓!

We noticed that this PR is missing a `RELEASE.md` file. We use that to automatically do releases here on GitHub and, most importantly, to PyPI!

So as soon as this PR is merged, a release will be made 🚀.

Here's an example of `RELEASE.md`:

```markdown
Release type: patch

Description of the changes, ideally with some examples, if adding a new feature.
```

Release type can be one of patch, minor or major. We use [semver](https://semver.org/),so make sure to pick the appropriate type. If in doubt feel free to ask :)

**Comment by botberry** [comment]
Hi, thanks for contributing to Strawberry 🍓!

We noticed that this PR is missing a `RELEASE.md` file. We use that to automatically do releases here on GitHub and, most importantly, to PyPI!

So as soon as this PR is merged, a release will be made 🚀.

Here's an example of `RELEASE.md`:

```markdown
Release type: patch

Description of the changes, ideally with some examples, if adding a new feature.
```

Release type can be one of patch, minor or major. We use [semver](https://semver.org/),so make sure to pick the appropriate type. If in doubt feel free to ask :)

**Review by patrick91 — APPROVED** [review]
Nice one! I'll check the pre-commit error and then merge ;)

File: docs/guides/errors.md [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+}
+---
+{
+  "data": null,
+  "errors": [
+    {
+      "message": "Can't find user",
+      "locations": [
+        {
+          "line": 3,
```
Comment (mecampbellsoup) [inline docs/guides/errors.md:119]: @jkimbo shouldn't this be `"line": 2`?

image

**Comment by mecampbellsoup** [comment]
This reads very nicely, good job @jkimbo!

File: docs/guides/errors.md (hunk 1 above)
Comment (jkimbo) [reply docs/guides/errors.md:119]: Well spotted! I'll change it.

Comment (jkimbo) [reply docs/guides/errors.md:119]: https://github.com/strawberry-graphql/strawberry/pull/860

**Pull request merged**